### ISSUE TITLE
govet: %r has unknown verb r for structs with custom formatters

### ISSUE THREAD

**Issue by ehotinger** [golangci/golangci-lint#414]
Here's an example. Running `go vet` for the same file does not produce any errors, but through `golangci-lint` it does. What's the expectation for this case?

```golang
const (
	escape = "\x1b"
	reset  = escape + "[0m"
	green  = escape + "[32m"
)

// Bar is a progress bar.
type Bar float64

var _ fmt.Formatter = Bar(1.0)

// Format the progress bar as output
func (h Bar) Format(state fmt.State, r rune) {
	switch r {
	case 'r':
	default:
		panic(fmt.Sprintf("%v: unexpected format character", float64(h)))
	}

	if h > 1.0 {
		h = 1.0
	}

	if h < 0.0 {
		h = 0.0
	}

	if state.Flag('-') {
		h = 1.0 - h
	}

	width, ok := state.Width()
	if !ok {
		// default width of 40
		width = 40
	}

	var pad int

	extra := len([]byte(green)) + len([]byte(reset))

	p := make([]byte, width+extra)
	p[0], p[len(p)-1] = '|', '|'
	pad += 2

	positive := int(Bar(width-pad) * h)
	negative := width - pad - positive

	n := 1
	n += copy(p[n:], green)
	n += copy(p[n:], bytes.Repeat([]byte("+"), positive))
	n += copy(p[n:], reset)

	if negative > 0 {
		copy(p[n:len(p)-1], bytes.Repeat([]byte("-"), negative))
	}

	_, _ = state.Write(p)
}

func main() {
	var b Bar = 0.9
	_, _ = fmt.Fprintf(os.Stdout, "%r", b)
}

```

1. Version of golangci-lint: `golangci-lint --version` (or git commit if you don't use binary distribution)
- `golangci-lint has version 1.15.0 built from 901cf25 on 2019-02-18T08:24:43Z`
2. Config file: `cat .golangci.yml`
- Default configuration.
3. Go environment: `go version && go env`
- go version go1.12 linux/amd64 -- nothing special about the env.
4. Verbose output of running: `golangci-lint run -v`

```
INFO [lintersdb] Active 10 linters: [deadcode errcheck gosimple govet ineffassign staticcheck structcheck typecheck unused varcheck] 
INFO [lintersdb] Optimized sublinters [staticcheck gosimple unused] into metalinter megacheck 
INFO [loader] Go packages loading at mode load deps types and syntax took 892.043523ms 
INFO [loader] SSA repr building timing: packages building 6.147839ms, total 48.898821ms 
INFO [runner] worker.1 took 449.638µs with stages: ineffassign: 400.462µs, deadcode: 45.035µs 
INFO [runner] worker.3 took 1.528392ms with stages: errcheck: 1.330878ms, varcheck: 164.279µs, structcheck: 23.905µs, typecheck: 1.805µs 
INFO [runner] worker.4 took 1.886151ms with stages: govet: 1.877959ms 
main.go:72:9: Fprintf format %r has unknown verb r (govet)
        _, _ = fmt.Fprintf(w, "%r", b)
               ^
```

**Comment by jirfag** [comment]
Hi, thank you!
I've merged #443 with the newest go vet. Now it works. I've added your program as a test case.
